python/ray/serve/common.py

```diff
@@ -35,13 +36,24 @@ def __init__(self,
         self.replica_config = replica_config
         # The time when .deploy() was first called for this deployment.
         self.start_time_ms = start_time_ms
-        self.actor_def = actor_def
+        self.actor_name = actor_name
+        self.serialized_deployment_def = serialized_deployment_def
         self.version = version
         self.deployer_job_id = deployer_job_id
         # The time when this deployment was deleted.
         self.end_time_ms = end_time_ms
         self.autoscaling_policy = autoscaling_policy
 
+    @property
+    def actor_def(self):
```

Comment:
what's the difference between putting it here as delayed import and init verses https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/serve/backend_state.py?L173:45 right before where actor_def is actually used in `start` ?

Reply:
no difference honestly. just keeping the code change footprint low.

Reply:
What's the reason for this change? We shouldn't need the `actor_def` once the replica is already created, right?

Reply:
Oh is it because we can't serialize the actor definition in the target_state and need to serialize the raw class definition instead?

This change means we're creating one actor definition per replica, which isn't ideal -- any way to still do it per-deployment?

Reply:
made it a cached property